src/flows/braze/update_flow.py

```diff
@@ -438,14 +448,16 @@ def mask_email_domain_outside_production(rows: List[Dict], email_column='EMAIL')
         track_users_task,  # Users creation needs to happen first, because otherwise deleted users might be recreated.
     )
 
-    # Set KV-store 'last_loaded_at' key to the maximum loaded_at seen so far if all tasks finished successfully.
-    set_last_loaded_at(all_user_deltas).set_dependencies(upstream_tasks=[
-        create_email_aliases_task,
-        delete_users_results,
-        identify_users_task,
-        subscribe_users_task,
-        track_users_task,
-    ])
+    # When doing a backfill (the table name is not the default), don't write loaded_at to the KV store.
```

Comment:
So far, we've only performed backfills from different tables. Do you foresee a use-case where we want to do a backfill from the same table, and not write to the KV store?

Predicting future requirements is usually hard, so I'd try to focus on meeting our existing use-cases.

Reply:
Even if we could distinguish between a manual and a scheduled run, I don't think it's safe to assume we don't want to write `loaded_at` to the KV store for all manual runs. For example, if a scheduled run failed, we might run the flow manually with default parameters, and I wouldn't expect the behavior to change because of that.

Reply:
> Even if we could distinguish between a manual and a scheduled run, I don't think it's safe to assume we don't want to write `loaded_at` to the KV store for all manual runs. For example, if a scheduled run failed, we might run the flow manually with default parameters, and I wouldn't expect the behavior to change because of that.

You have a good point. It also means having a few checks:
For `manual` _not_ backfill:
- `loaded_at` should be updated
- scheduled runs should be temporarily disabled

For `backfill`:
- `loaded_at` should _not_ be updated

How about having a first parameter expecting an input from the user if its going to be a `manual` or a `backfill` run with the necessary warnings? Based on the users input determine how the run plays out?

Reply:
> How about having a first parameter expecting an input from the user if its going to be a manual or a backfill run with the necessary warnings? Based on the users input determine how the run plays out?

It's not possible (as far as I know) to require user input when manually running a flow.

One alternative would be to explicitly add a boolean parameter `write_loaded_at_to_kv_store` that is True by default. The risk I see there is that someone will forget to disable it during a backfill.

Reply:
I don't see a simple and safe way to have two flows, although I agree that would be a better solution, if possible. Duplicated code (as we have now) runs the risk of diverging, when someone forgets to make a fix in both places. I made [a note](https://docs.google.com/document/d/1RFFgnBGWBOjnRvonWgQGXqolN8wUGsQzdZ_QcEsWhXs/edit) to ask Prefect during our 90 day sync next month for a recommendation on how to extend flows.